### ISSUE TITLE
FunctionClauseError following a connection redirect

### ISSUE THREAD

**Issue by muelthe** [elixir-ecto/tds#138]
### Environment

* Elixir version (elixir -v): 1.13.4
* TDS version (mix deps): 2.3.0
* MSSQL Server version (SQL Server 2016 / 2019 / Azure SQL Edge): 2014 -  Microsoft SQL Azure (RTM) - 12.0.2000.8
* Additional configuration parameter (`ssl_opts`, `socket_options`, ...): `[ssl: true, show_sensitive_data_on_connection_error: true]`
* Operating system: Windows/Linux

### Expected behavior
Running: `TDS.Query(pid, query, [])`
Expected: `{:ok,  %Tds.Result{...}}`

### Actual behavior
**Note:** running with `show_sensitive_data_on_connection_error: true`

```
[error] GenServer #PID<0.1478.0> terminating
** (FunctionClauseError) no function clause matching in String.length/1
    (elixir 1.13.4) lib/string.ex:1968: String.length(:REDACTED)
    (tds 2.3.0) lib/tds/protocol/login7.ex:131: Tds.Protocol.Login7.encode_variable_login/2
    (tds 2.3.0) lib/tds/protocol/login7.ex:92: Tds.Protocol.Login7.encode/1
    (tds 2.3.0) lib/tds/protocol.ex:814: Tds.Protocol.login_send/2
    (tds 2.3.0) lib/tds/protocol.ex:530: Tds.Protocol.login/1
    (tds 2.3.0) lib/tds/protocol.ex:369: Tds.Protocol.connect/2
    (tds 2.3.0) lib/tds/protocol.ex:483: Tds.Protocol.decode/2
    (tds 2.3.0) lib/tds/protocol.ex:530: Tds.Protocol.login/1
Last message: nil
State: Tds.Protocol
** (EXIT from #PID<0.1442.0>) shell process exited with reason: killed
```

### Troubleshooting
I understand the following occurs, leading to the error.

#### Summary of connection
* the connection is setup with all resoiurces in the same Azure tenancy
* DNS responds with the public IP
* a Service Endpoint in Azure redirects the connection to a local address (i.e. on the same vnet)
* `Tds.Protocol.message/3` is called matching on `:redirect`
* New hostname and port options are set and `connect(new_opts)` is called

There is a private function `Tds.Protocol.clean_opts/1` which is called before the redirect takes place. i.e. the password field is overwritten with the atom `:REDACTED`, which in turn leads to the error (`String.length/1` expects a string/binary).

### Possible solution
`Tds.Protocol.clean_opts/1` is called twice, once from `Tds.Protocol.login_send/4` and in `Tds.Protocol.message/3` where status matches on `:login`

If I am following the code correctly, I believe the call to `Tds.Protocol.clean_opts/1` could be moved to the the `:disconnect` clause of the case statement in `Tds.Protocol.login_send/4`. This would ensure the password field is redacted, as intended, for both a connect and disconnect.

If you're happy with this approach, I would be happy to create a new PR and complete as proposed - alternatively I would be happy to assist in any other way.

**Comment by moogle19** [comment]
Hi @muelthe 
I just pushed a commit with your intended fix.
Could you give the `master` branch a try and check if it fixes you problem?

**Comment by muelthe** [comment]
Hi @moogle19 
I can confiirm that this fixes the problem.
Thank you ever so much!

**Comment by moogle19** [comment]
@muelthe 
Just released a new version (`2.3.1`) with the fix on hex.pm!
